### DISCUSSION TITLE
24.10.2021

### DISCUSSION THREAD

**Discussion by farag2** [farag2/Sophia-Script-for-Windows#276] [General]
# Before running the script read [Before running](https://github.com/farag2/Sophia-Script-for-Windows#fire-before-running-fire) & [How to use](https://github.com/farag2/Sophia-Script-for-Windows#how-to-use)

[![YT](https://img.youtube.com/vi/f529ucAipI8/0.jpg)](https://youtu.be/f529ucAipI8) [![YT](https://img.youtube.com/vi/MiQ85tVXQQA/0.jpg)](https://youtu.be/MiQ85tVXQQA)

***

[![ko-fi](https://www.ko-fi.com/img/githubbutton_sm.svg)](https://ko-fi.com/Q5Q51QUJC)

<a href="https://yoomoney.ru/to/4100116615568835"><img src="https://yoomoney.ru/i/shop/iomoney_logo_color_example.png" width=220px height=46px></a>

***

[![actions](https://img.shields.io/badge/Sophia%20News-Telegram-blue?style=flat&logo=Telegram)](https://t.me/SophiaNews) [![actions](https://img.shields.io/badge/Sophia%20Chat-Telegram-blue?style=flat&logo=Telegram)](https://t.me/Sophia_Chat) [![actions](https://img.shields.io/github/workflow/status/farag2/Sophia-Script-for-Windows/Build?label=GitHub%20Actions&logo=GitHub)](https://github.com/farag2/Sophia-Script-for-Windows/actions)

## Windows 11 21H2 | Windows 10 2004/20H2/21H1/21H2 | LTSC 1809

Diff from v5.12.4
[5.12.4...5.12.5](https://github.com/farag2/Sophia-Script-for-Windows/compare/5.12.4...5.12.5)

* Now you can download the always latest Sophia Script archive by invoking (`not as administrator too`) in PowerShell console

```powershell
irm script.sophi.app | iex
```

or without using aliases

```powershell
Invoke-RestMethod -Uri script.sophi.app | Invoke-Expression
```

* The command will download and expand the archive (`without running`) the latest Sophia Script according which Windows and PowerShell versions it is run on. If you run it on Windows 11 via PowerShell 5.1, it will download Sophia Script for `Windows 11 PowerShell 5.1`.

* The `NotificationAreaIcons`, `TaskbarSize`, `RecentlyAddedAppsAdded` funtions were removed due to lack of necessity;
* Fixed typo in `FileExplorerCompactMode`;
  * The values for `Disable` and `Enable` were switched places by mistake;
* Fixed bug in `UpdateMicrosoftProducts`;
* Fixed bug in `IPv6Component`;
  * The parameter uses `U+2013` instead of `U+2010` character (hyphen)
* Fixed major in `WinPrtScrFolder` (closes #260);
  * It turned out that it has an influence on the OneDrive behavoir. Now the function will be applied only if the preset is configured to remove OneDrive, otherwise the backup functionality for the "Desktop" and "Pictures" folders in OneDrive breaks;
* Merged #264;
* Added `Spotify` for removing to `UninstallUWPApps`;
* `ShareContext` now uses another method that doesn't have an influence on `Windows10FileExplorer`
  * To revert changes invoke

  ```powershell
  Remove-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Shell Extensions\Blocked" -Name "{E2BF9676-5F8F-435C-97EB-11607A5BEDF7}" -Force -ErrorAction Ignore
  ```

  * And then invoke a new `ShareContext` [(how-to)](https://github.com/farag2/Sophia-Script-for-Windows#how-to-run-the-specific-functions);
* `OpenWindowsTerminalAdminContext` rewritten and now uses a native UAC icon and added to the top-level context menu
  * To revert changes invoke

  ```powershell
  $Items = @(
    "Registry::HKEY_CLASSES_ROOT\Directory\Background\shell\OpenWTHereAsAdmin",
    "Registry::HKEY_CLASSES_ROOT\Directory\shell\OpenWTHereAsAdmin",
    "HKLM:\SOFTWARE\Classes\Directory\shell\OpenWTHereAsAdmin\command",
    "HKLM:\SOFTWARE\Classes\Directory\Background\shell\OpenWTHereAsAdmin\command",
    "$env:LOCALAPPDATA\Packages\Microsoft.WindowsTerminal_8wekyb3d8bbwe\LocalState\terminal.ico"
  )
  Remove-Item -Path $Items -Recurse -Force -ErrorAction Ignore
  ```

  ![Context](https://i.imgur.com/3xUXRQJ.png)

  * And then invoke a new `OpenWindowsTerminalAdminContext` [(how-to)](https://github.com/farag2/Sophia-Script-for-Windows#how-to-run-the-specific-functions);
* Added function to set the Windows 10 style context menu (closes #267);
* Minor changes.
* Check out the 3rd [SophiApp](https://github.com/Sophia-Community/SophiApp) public alpha build :rocket:
  * The 4th alpha version 0.0.0.5x will be released soon

![SophiApp](https://hsto.org/r/w780/getpro/habr/upload_files/be9/060/0b6/be90600b648639aa85d755fe10677cb2.jpg)

## Sophia Script Wrapper 2.5.1

* Cleaned up code: EXE should be now smaller;
* Fixed typos in JSONs;
* Added Spanish (es-ES);
* Fixed UI.

![Wrapper](https://i.imgur.com/DRRQPJh.png)

|              File                                     |                          SHA-256                               |
|:-----------------------------------------------------:|:--------------------------------------------------------------:|
| Sophia.Script.for.Windows.10.LTSC.v5.2.16.zip         |368A6A4588749999A31000341A176E2615B93F8A2C0AB4DF91535DE95C064FD6|
| Sophia.Script.for.Windows.10.v5.12.5.PowerShell.7.zip |9BDDD05FFFAEC24490E345F947C37C9665A3E4E3C86F3B6FE2EAC6FABB1C3F57|
| Sophia.Script.for.Windows.10.v5.12.5.zip              |C10ACE107FB0CA48297DCF3C8E195F5DE0E0224762C2CD616D3740A38B3B8F1E|
| Sophia.Script.for.Windows.11.v6.0.6.PowerShell.7.zip  |CF3A955A34FB19BB67F8959620E694ADA2B6B6590FEBF294F453EB07460B9054
| Sophia.Script.for.Windows.11.v6.0.6.zip               |C238A1A23675B2C64B17408E8208660D908110C7BA0D28D939E4CC1A14D4DDCF|
|  Sophia.Script.Wrapper.v2.5.1.zip                     |F66A6CF7760C2E174790FC83F624F5026E10EA9EBCC4F8B7253933444D8D491A|

<hr /><em>This discussion was created from the release <a href='https://github.com/farag2/Sophia-Script-for-Windows/releases/tag/6.0.6'>24.10.2021</a>.</em>

**Comment by Temtaime** [comment]
But who said there's no necessity for NotificationAreaIcons, etc ?
It is time-consuming leaving those functions intact ?

**Reply by farag2** [reply]
What?..

**Reply by Temtaime** [reply]
I was using NotificationAreaIcons and TaskbarSize functions. And now i'm forced not to update the script because one said that there's no necessity for them. Or should i use multiple versions for multiple functions ? How is it supposed to be used ?

**Reply by farag2** [reply]
NotificationAreaIcons doesn't work any more... And the TaskbarSize function uses a non-documented feature. I decided to remove it due to it actually makes the taskbar ugly. But you may add the function to the latest release by yourself. This.is very simple if you still like the function.
